tchannel/tornado/message_factory.py

```diff
@@ -239,94 +236,120 @@ def build_response(self, message):
         )
         return res
 
-    def build_context(self, message):
-        if message.message_type == Types.CALL_REQ:
-            return self.build_request(message)
-        elif message.message_type == Types.CALL_RES:
-            return self.build_response(message)
+    def build_inbound_response(self, message, response):
+        """buffer all the streaming messages based on the
+        message id. Reconstruct all fragments together.
+
+        :param message:
+            incoming message
+        :param response:
+            incoming response
+        :return: next complete message or None if streaming or fragmentation
+            is not done
+        """
+        if message.message_type == Types.CALL_RES:
+            self.verify_message(message)
+            response = self.build_response(message)
+            num = self._find_incompleted_stream(response)
+            self.close_argstream(response, num)
+            return response
+
+        elif message.message_type == Types.CALL_RES_CONTINUE:
+            if response is None:
+                # missing call msg before continue msg
+                raise FatalProtocolError(
+                    "missing call message after receiving continue message")
 
-    def build(self, message):
+            dst = self._find_incompleted_stream(response)
+            try:
+                self.verify_message(message)
+            except InvalidChecksumError as e:
+                response.argstreams[dst].set_exception(e)
+                raise
+
+            src = 0
+            while src < len(message.args):
+                response.argstreams[dst].write(message.args[src])
+                dst += 1
+                src += 1
+
+            if message.flags != FlagsType.fragment:
+                # get last fragment. mark it as completed
+                assert (len(response.argstreams) ==
+                        CallContinueMessage.max_args_num)
+                response.flags = FlagsType.none
+
+            self.close_argstream(response, dst - 1)
+            return None
```

Comment:
add a catch-all else which raises an exception or something if this should never be called with anything besides `CALL_RES*`